website/static/js/registrationEmbargo.js

```diff
@@ -0,0 +1,89 @@
+var ko = require('knockout');
+
+var RegistrationEmbargoViewModel = function() {
+
+    var MAKE_PUBLIC = 'Make registration public immediately';
+    var MAKE_EMBARGO = 'Enter registration into embargo';
+    var self = this;
+
+    var today = new Date();
+    var TWO_DAYS_FROM_TODAY_TIMESTAMP = new Date().getTime() + (2 * 24 * 60 * 60 * 1000);
+    var ONE_YEAR_FROM_TODAY_TIMESTAMP = new Date().getTime() + (365 * 24 * 60 * 60 * 1000);
+
+    self.dayChoice = ko.observable();
+    self.monthChoice = ko.observable();
+    self.yearChoice = ko.observable();
+
+    self.registrationOptions = ko.observableArray([
+        MAKE_PUBLIC,
+        MAKE_EMBARGO
+    ]);
+    self.registrationChoice = ko.observable(MAKE_PUBLIC);
+
+    self.dayOptions = ko.computed(function() {
+        var num_of_days;
+
+        if (['January', 'March', 'May', 'July', 'August', 'October', 'December'].indexOf(self.monthChoice()) > -1) {
+            num_of_days = 31;
+        } else if (['April', 'June', 'September', 'November'].indexOf(self.monthChoice()) > -1) {
+            num_of_days = 30;
+        } else { num_of_days = 28; }
+
+        var days = [];
+        for (var i=1; i<=num_of_days; i+=1) {
+            days.push(i);
+        }
+        return days;
+    });
+    self.monthOptions = ko.observableArray([
+        'January',
+        'February',
+        'March',
+        'April',
+        'May',
+        'June',
+        'July',
+        'August',
+        'September',
+        'October',
+        'November',
+        'December'
+    ]);
+    self.yearOptions = ko.computed(function() {
+        return [today.getFullYear(), today.getFullYear() + 1];
+    });
+    self.showEmbargoDatePicker = ko.observable(false);
+    self.checkShowEmbargoDatePicker = function() {
```

Comment:
Can move all functions to the prototype of RegistrationEmbargoViewModel, i.e.
```RegistraionEmbargoViewModel.prototype.checkShowEmbargoDatePicker```,

but I think maybe this could/should be a ko.pureComputed?